src/pymovements/dataset/dataset.py

```diff
@@ -159,6 +158,8 @@ def load(
                 events_dirname=events_dirname,
                 extension=extension,
             )
+            for loaded_gaze_df, loaded_events_df in zip(self.gaze, self.events):
+                loaded_gaze_df.events = loaded_events_df
```

Comment:
I'm not sure if this is really necessary, but better to be safe than sorry.

I created #1110 to resolve the underlying issue.